File: cs50/grocery.py
# append to list only when a unique item is captured. sort alphabatically

groceryDict = {}

while (True):

    try:
        grocery = input()
        
        if grocery not in groceryDict:
            groceryDict[grocery]=1
        elif grocery in groceryDict:
            
            groceryDict[grocery]+=1

    except EOFError:

        sortedkeys=sorted(groceryDict.keys())
        
        for i in sortedkeys:
            print(f"{groceryDict[i]}.{i.upper()}")
                    # here we are printing the value of i in the dict, here after sorting the i's are stored in a list, so t access the i's dictionary content we can access grocerydict of i where the 'i' key has a certain value which is the count, refer above code , printing the value associated with the key->i.
        break

l = [1,2]
sorted(l)

print(l)

File: cs50/grocery.c
# include <stdio.h>

typedef struct grocery{
    int count;
    char firstLetter;
}item;

int main(){
    item a[9];
    
    char letter;
    int i=-1,k=0;
    int j=0;
    for (i=0;i<9;i++){
        a[i].count = 1;
    }
     i = 0;
    while(i<9-1){
        int flag = 0;
        printf("item: ");
        scanf(" %c",&letter);
        
        if (letter == '|'){
            break;
        }else{
            for (j=0;j<9;j++){
                if (a[j].firstLetter == letter ){
                    a[j].count++;
                    int flag = 1;
                    break;
                }
            }
            if (flag != 1){
                i++;
                a[i].firstLetter = letter;
                a[i].count = 1;
                
            }
        } 

        

    }
    k=i;

    for (i=0;i<k;i++){
        printf(" %c\t",a[i].firstLetter);
        printf("%d",a[i].count);
        printf("\n");
    }

}